### DISCUSSION TITLE
Opentelemetry integration with reverse-proxy (haproxy and AWS ELB)

### DISCUSSION THREAD

**Discussion by prakash-42** [open-telemetry/community#973] [Community Discussions]
Is there a solution to have open-telemetry integrate with reverse-proxy? We use a combination of haproxy and AWS Elb for routing of our requests.

I have already seen a solution for nginx (https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/nginx), so I'm hoping there could be a similar solution for haproxy and AWS ELB.

**Comment by SergeyKanzhelev** [comment]
Are you interested in traces, logs, or metrics?

CC @alolita @anuraaga for AWS ELB portion of this question

For haproxy short googling uncovered things like:

https://gist.github.com/wosc/dcda023a11093263f1febc7a050b949e and https://docs.splunk.com/observability/gdi/haproxy/haproxy.html

@bogdandrutu do you know whether Splunk solution is based on some OSS component that can be reused?

@wosc I wonder if you have managed to convert the solution written in gist in working OSS component?

**Reply by prakash-42** [reply]
@SergeyKanzhelev I'm interested in traces (And putting trace ID in logs if possible). I did find https://gist.github.com/wosc/dcda023a11093263f1febc7a050b949e , but wasn't sure if I can go ahead with POC (as I'm using open-telemetry + jaeger; while this was open-telemetry + zipkin). Will read more and try to understand @wosc's solution.

**Reply by wosc** [reply]
To clarify: I judge the fluentbit+otelcol approach I've sketched in that gist to be more of a workaround. Yes, it does work (we are sucessfully using it in production), but it feels somewhat cumbersome to me (requires fluentbit as an additional component, requires evaluating lua code in fluentbit to tweak some string formatting _just right_, and so on).

From my perspective, a "proper" integration would be either for otelcol to gain functionality to directly process old-fashioned logfiles into tracing data, or to create a component that leverages the modern haproxy [SPOA](https://www.haproxy.com/blog/extending-haproxy-with-the-stream-processing-offload-engine/) functionality to generate otlp protocol data directly (and perform header propagation as well if that's possible).

**Comment by vikasjio** [comment]
Is there any progress on 1. haproxy generating Opentelemetry compliant traces itself or with hejp of some tool/libs? and 2. availability of collector (or at least receiver) for the same?

**Reply by svrnm** [reply]
@vikasjio this is a question worth asking the haproxy community, since the best way to enable opentelemetry compatibility is by doing a direct integration, e.g. there is a long standing issue: https://github.com/haproxy/haproxy/issues/1640
